indexer/index.py

```diff
@@ -308,68 +282,60 @@ def validate_path(path: str) -> None:
     raise Exception(f"Invalid path ({message}).")
 
 
-def index(
-    client_secret: str,
-    site: str,
-    paths: Optional[List[str]] = defaults.PATHS,
-    paths_file: str = defaults.PATHS_FILE,
-    blacklist: Optional[List[str]] = defaults.BLACKLIST,
-    blacklist_file: str = defaults.BLACKLIST_FILE,
-    address: str = defaults.ADDRESS,
-    timeout: int = defaults.TIMEOUT,
-    retries: int = defaults.RETRIES,
-    basic_only: bool = defaults.BASIC_ONLY,
-    patch: bool = defaults.PATCH,
-    iceprodv2_rc_token: str = defaults.ICEPRODV2_RC_TOKEN,
-    iceprodv1_db_pass: str = defaults.ICEPRODV1_DB_PASS,
-    dryrun: bool = defaults.DRYRUN,
-    non_recursive: bool = defaults.NON_RECURSIVE,
-    n_processes: int = defaults.N_PROCESSES,
-    token_url: str = defaults.TOKEN_URL,
-    client_id: str = defaults.CLIENT_ID,
-) -> None:
+def index(args: Namespace) -> None:
```

Comment:
This removes the feasibility to import and call `index()`, see https://github.com/WIPACrepo/file-catalog-indexer#from-indexerindex-import-index. It cleans the code up nicely, but it's something worth deciding now on how we're going to use the tool. You're the expert, so if we're not using the importability, let's nix it and update the README.

Reply:
I'll make some changes to preserve importability.
My specific use cases (lta and retro-disk-copy) use it through a subprocess rather than directly.
However, that doesn't mean somebody won't want to in the future.